contract/models/account_invoice.py

```diff
@@ -10,3 +10,9 @@ class AccountInvoice(models.Model):
     contract_id = fields.Many2one(
         'account.analytic.account', string='Contract'
     )
+
+    @api.multi
+    def finalize_creation_from_contract(self):
+        for invoice in self:
+            invoice._onchange_partner_id()
+        self.compute_taxes()
```

Comment:
This method is a bit weird here alone on account invoice. BTW, does it need to be public?

Reply:
@sbidoul I can make it `@api.model` and move it to the contract model. It was not supposed to be public, it's a mistake.